pipeline.ipynb

```diff
@@ -2,606 +2,618 @@
+    "\n",
+    "To mask our grid to the forested area of Alaska, we want a forest mask raster that is on the same grid as our new datasets. So we want to create a GeoTIFF file for the NCAR grid as a template.\n",
+    "\n",
+    "Read in one of the time slices of an NCAR file to get the grid (i.e. just the 2-D array of data), derive the projection info using some info provided by NCAR about this dataset, reproject to EPSG:3338 and create a GeoTIFF template file:"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 301,
-   "id": "6c5cdfc8-90bc-4cb6-a8a9-59ca979492af",
+   "execution_count": 2,
+   "id": "883fbe2f",
    "metadata": {},
    "outputs": [],
    "source": [
-    "risk_dict = {\"low\": {}, \"high\": {}}"
+    "# open an NCAR file to get some info from\n",
+    "with xr.open_dataset(met_dir.joinpath(\"daymet/daymet_met_1980.nc\")) as ds:\n",
+    "    # need grid shape below\n",
+    "    ny, nx = ds.longitude.shape\n",
+    "    ncar_arr = np.flipud(ds[\"tmin\"].values[0])\n",
+    "\n",
+    "# values provided by NCAR (via email correspondence)\n",
+    "wrf_proj_str = PolarStereographic(**{\"TRUELAT1\": 64, \"STAND_LON\": -150}).proj4()\n",
+    "wrf_proj = Proj(wrf_proj_str)\n",
+    "wgs_proj = Proj(proj='latlong', datum='WGS84')\n",
+    "transformer = Transformer.from_proj(wgs_proj, wrf_proj)\n",
+    "e, n = transformer.transform(-150, 64)\n",
+    "# Grid parameters\n",
+    "dx, dy = 12000, 12000\n",
+    "# Down left corner of the domain\n",
+    "x0 = -(nx-1) / 2. * dx + e\n",
+    "y0 = -(ny-1) / 2. * dy + n\n",
+    "# 2d grid\n",
+    "x = np.arange(nx) * dx + x0\n",
+    "y = np.arange(ny) * dy + y0\n",
+    "\n",
+    "# these coordinates will be used here and for spatially\n",
+    "#  referencing all resulting risk class data files\n",
+    "ncar_coords = {\n",
+    "    \"y\": ([\"y\"], np.flip(y)),\n",
+    "    \"x\": ([\"x\"], x),\n",
+    "}\n",
+    "\n",
+    "da = xr.DataArray(\n",
+    "    data=ncar_arr,\n",
+    "    dims=[\"y\", \"x\"],\n",
+    "    coords=ncar_coords,\n",
+    ")\n",
+    "da.attrs[\"_FillValue\"] = np.nan\n",
+    "\n",
+    "# temp_ncar_fp = scratch_dir.joinpath(\"ncar_template_3338.tif\")\n",
+    "da.rio.write_crs(wrf_proj_str).rio.reproject(\"EPSG:3338\").rio.to_raster(temp_ncar_fp)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 302,
-   "id": "3af9b569-9436-4f43-843b-d78512d9719e",
+   "cell_type": "markdown",
+   "id": "995291dc",
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "114 s\n"
-     ]
-    }
-   ],
    "source": [
-    "import time\n",
+    "Note, we are reprojecting to EPSG:3338 because it is generally going to be more user friendly than the current grid, especially for ingesting into Rasdaman. \n",
+    "\n",
+    "#### 2.1.2 - Regrid the forest mask\n",
```

Comment:
I ran into this same problem and dug into it a bit. When I encountered the warning @charparr mentioned above, the `risk_class` GeoTIFFs that were produced had good data, but did not have the correct projection (`EPSG:3338`) according to `gdalinfo` and QGIS. I traced my steps and found a way to fix this, but it's a little strange.

When I first ran the notebook, I created a new conda environment I called "beetles" with `anaconda-project` installed so I could open the pipeline notebook properly:

```
conda activate beetles
anaconda-project run pipeline
```

After the above command finished and I opened the notebook in my browser, running the cells produced the warnings starting at the same location as @charparr, 2.1.2.

But I noticed that, in the process of running the `anaconda-project run pipeline` command above, a new conda environment was created in the repo's directory. I discovered if I activated this environment and then opened the notebook again, like so:

```
conda activate ~/repos/spruce-beetle-risk/envs/default
anaconda-project run pipeline
```

And then stepped through the cells again, the warning went away and produced `risk_class` GeoTIFFs with the correct `EPSG:3338` projection.

This is tricky because it's a chicken and egg kind of situation. I had to run the notebook once with my own `beetles` conda environment to get it to create the `default` conda environment that ultimately worked. I'm not sure how to solve this in a more straightforward way, but I suppose just adding some instructions to the README to activate to the `spruce-beetle-risk/envs/default` environment once it is created can suffice until we find a simpler way.

Reply:
Gosh, this is anonying - I think I am going to just refactor to have an `environment.yml` file and update the README with instructions. I don't think `anaconda-project` is really worth it right now. It's more straightforward to not have the env-within-env situation - it's caused trouble for me in the past as well.